solver/src/orderbook.rs

```diff
@@ -21,19 +27,29 @@ impl OrderBookApi {
         url.set_path(PATH);
         self.client.get(url).send().await?.json().await
     }
+
+    pub fn get_native_token(&self) -> WETH9 {
+        self.native_token.clone()
+    }
```

Comment:
What's the overhead here of cloning the actual Contract instance vs passing around an Arc?

Reply:
They are lightweight. The actual ABI is kept in a `lazy_static` object so that cloning just means cloning the transport (which, AFAIK, boils down to cloning an `Arc`).